### DISCUSSION TITLE
Need help regarding SLSQP optimizer algorithm and how it uses sensitivity data from ADJOINT solver

### DISCUSSION THREAD

**Discussion by usmanshahid90** [mdolab/dafoam#133] [General]
Hi everyone, 
                   I am trying to develop a flow diagram of SLSQP optimization algorithm and want to link it with the sensitivity derivatives coming from the adjoint solver.  I am unable to figure out that how the sensitivity derivatives computed from the adjoint solver are to be utilized in SLSQP algorithm.  Moreover, if anyone can guide me about SLSQP, I would be really thankful. 

Kindly guide me in this regard.
Thanks.

**Comment by friedenhe** [comment]
This is done in pyOptSparse. Check this [example](https://github.com/mdolab/pyoptsparse/blob/master/examples/rosenbrock.py).
